bin/storm.py

```diff
@@ -541,6 +541,13 @@ def initialize_upload_credentials_subcommand(subparsers):
         help="""name of the owner of the topology (security precaution)"""
     )
 
+    # If set, this flag will become true meaning that user expects non-empty creds to be uploaded.
+    # Command exits with non-zero code if uploaded creds collection is empty.
+    sub_parser.add_argument(
+        "-e", "--exception", action='store_true',
```

Comment:
`--exception` is too general. Maybe `--fail-on-empty` or something

Reply:
I changed --exception-when-empty here since we already use short name `-f` for file.

Reply:
This is also good

Reply:
Where is args.exception_when_empty passed to java command (in storm.py:upload_credentials()?

Reply:
@bipinprasad https://github.com/apache/storm/blob/088e722559437cbd55ae7073c7fb89ac0f8c4ea5/storm-core/src/jvm/org/apache/storm/command/UploadCredentials.java#L45

Reply:
Seems there is a slight problem. In python code, when exec_java_class() is called, the arguments to the java class is constructed partly from main_class_args. main_class_args is constructed by simply eliminating some arguments passed to the python code (and keeping the rest intact). So if python code got passed "-e" ii will get passed thru to java class as "-e". However, if "--exception was passed to python, then --exception will get passed to java. This wont match what java class is expecting. So in order to do a passthru, the option names should be the same, otherwise main_class_args should be suitably modified based on the args. Looks like passthru might be preferred solution.

Reply:
Yea, the argument in python is the same: "--exception-when-empty".